src/main/scala/com/github/gvolpe/fs2/Fifo.scala

```diff
@@ -1,43 +1,39 @@
 package com.github.gvolpe.fs2
 
-import cats.effect.{Effect, IO}
-import fs2.StreamApp.ExitCode
-import fs2.async.mutable.Queue
-import fs2.{Scheduler, Stream, StreamApp, async}
-
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.duration._
-
-object FifoApp extends Fifo[IO]
+import cats.effect._
+import fs2.Stream
+import fs2.concurrent.Queue
 
 /**
-  * Represents a FIFO (First IN First OUT) system built on top of two [[fs2.async.mutable.Queue]].
+  * Represents a FIFO (First IN First OUT) system built on top of two [[fs2.concurrent.Queue]].
   *
   * q1 has a buffer size of 1 while q2 has a buffer size of 100 so you will notice the buffering when
   * pulling elements out of the q2.
   * */
-class Fifo[F[_]: Effect] extends StreamApp[F] {
-
-  override def stream(args: List[String], requestShutdown: F[Unit]): fs2.Stream[F, ExitCode] =
-    Scheduler(corePoolSize = 4).flatMap { implicit S =>
-      for {
-        q1 <- Stream.eval(async.boundedQueue[F, Int](1))
-        q2 <- Stream.eval(async.boundedQueue[F, Int](100))
-        bp = new Buffering[F](q1, q2)
-        ec <- S.delay(Stream.emit(ExitCode.Success).covary[F], 5.seconds) concurrently bp.start.drain
-      } yield ec
-    }
+object FifoApp extends IOApp {
+  import cats.syntax.all._
+
+  def stream[F[_]: ConcurrentEffect]: fs2.Stream[F, Unit] =
+    for {
+      q1 <- Stream.eval(Queue.bounded[F, Int](1))
+      q2 <- Stream.eval(Queue.bounded[F, Int](100))
+      _ <- new Buffering[F](q1, q2).start
+    } yield ()
+
+  override def run(args: List[String]): IO[ExitCode] =
+    stream[IO].compile.drain.as(ExitCode.Success)
 
 }
 
-class Buffering[F[_]](q1: Queue[F, Int], q2: Queue[F, Int])(implicit F: Effect[F]) {
+class Buffering[F[_]](q1: Queue[F, Int], q2: Queue[F, Int])(
+    implicit F: Concurrent[F]) {
 
   def start: Stream[F, Unit] =
```

Comment:
Rename to `exec` and make it a `val`.

Reply:
Renamed but since I transformed `Buffering` to be an `object`, `exec` should stay a function. WDYT ?